registry/storage/vacuum.go

```diff
@@ -27,14 +27,20 @@ type Vacuum struct {
 	ctx    context.Context
 }
 
-// RemoveBlob removes a blob from the filesystem
-func (v Vacuum) RemoveBlob(dgst string) error {
+// RemoveBlob removes a blob from the filesystem. withDirectory
+// allows to remove <hex-digest> directory containing the blob as well.
+func (v Vacuum) RemoveBlob(dgst string, withDirectory bool) error {
```

Comment:
This interface addition makes no sense. What would removing a blob ever have to do with a directory. This leaks any notion of abstraction.

Also, the type here should be a digest, not a string.